src/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,194 @@
+ *
+ * * When it will take a long time to load/render all items.
+ * * If you want to browse the data by navigating through pages.
+ */
+
+export function Pagination({
+  defaultCurrent = 1,
+  total,
+  showSizeChanger = false,
+  showQuickJumper = false,
+}: PaginationProps) {
+  const [totalPages, setTotalPages] = useState(Math.ceil(total / 10))
+  const [selected, setSelected] = useState(defaultCurrent)
+  const [inputValue, setInputValue] = useState<string>('')
+
+  const options = [
+    { label: '10 / page', value: '10' },
+    { label: '20 / page', value: '20' },
+    { label: '50 / page', value: '50' },
+    { label: '100 / page', value: '100' },
+  ]
+
+  useEffect(() => {
+    if (selected > totalPages) {
+      setSelected(totalPages)
+    }
+  }, [selected, totalPages])
+
+  function handlePageSize(selectedOption: string | undefined) {
+    const newPageSize = parseFloat(selectedOption ?? '10')
+    setTotalPages(Math.ceil(total / newPageSize))
+  }
+
+  function handleDecrement() {
+    setSelected((prevState) => prevState - 1)
+  }
+
+  function handleIncrement() {
+    setSelected((prevState) => prevState + 1)
+  }
+
+  function handleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    const newValue = event.target.value
+    setInputValue(newValue)
+  }
+
+  function handlePageJump() {
+    const jumpValue = parseFloat(inputValue ?? '')
+    if (typeof jumpValue === 'number') {
+      if (jumpValue < 1) {
+        setSelected(1)
+      } else if (jumpValue > totalPages) {
+        setSelected(totalPages)
+      } else {
+        setSelected(jumpValue)
+      }
+    }
+    setInputValue('')
+  }
+
+  function handleEnterKeyPress(event: React.KeyboardEvent) {
+    if (event.key === 'Enter') {
+      handlePageJump()
+    }
+  }
+
+  const firstPageButton = (
+    <PageButton page={1} selected={selected} onClick={() => setSelected(1)} />
+  )
+
+  const renderButtons = useCallback(
+    (x, y) => {
+      return range(x, y).map((page) => (
+        <PageButton
+          key={page}
+          page={page}
+          selected={selected}
+          onClick={() => setSelected(page)}
+        />
+      ))
+    },
+    [selected],
+  )
+
+  const lastPageButton = (
+    <PageButton
+      page={totalPages}
+      selected={selected}
+      onClick={() => setSelected(totalPages)}
+    />
+  )
+
+  return (
+    <div className="flex items-center gap-x-2">
+      {/* Previous Button */}
+      <Button
+        label="Previous"
+        buttonType={Button.ButtonType.link}
+        icon={<ChevronLeftOutline />}
+        disabled={selected === 1}
+        onClick={handleDecrement}
+      />
+
+      {totalPages < 8 ? (
+        renderButtons(1, totalPages + 1)
+      ) : (
+        <>
+          {firstPageButton}
+          <span
+            className={clsx(
+              'text-blue-600',
+              selected > 4 ? 'inline' : 'hidden',
+            )}
+          >
+            ...
+          </span>
+          {selected < 4
```

Comment:
Instead of ternary operators, we can simplify this logic as

```jsx
let start
let end

if (totalPages - selected < 3) {
  start = totalPages - 3
  end = totalPages
}

renderButtons(start, end)
```